libs/Texture/Texture.h

```diff
@@ -0,0 +1,39 @@
+#ifndef TEXTURELOADER_H_
+#define TEXTURELOADER_H_
+
+#include <GL/glew.h>
+#include <string>
+
+class Texture
+{
+public:
+	Texture(GLenum TextureTarget, const std::string& filename);
+	~Texture();
+
+	void Bind(GLenum TextureUnit);
+	GLuint GetTextureId();
+
+	int LoadTexture(const std::string& fileName, int& width, int& height, unsigned char ** data, int& bbp);
```

Comment:
And I don't think ints should be pass by reference.. I think passing by reference maybe even slower for small values